Added a new test case to cleanUpPreviousRoundUnitTest + cleanUp

shine2lay opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/wetrustplatform/rosca-contracts/89)
<!-- Reviewable:end -->

Commit:
PR ready

Commit:
PR ready

Commit:
README fixed

Commit:
Merge branch 'develop' of github.com:WeTrustPlatform/rosca-contracts into main_develop_recently_pulled

Commit:
Merge branch 'develop' of github.com:WeTrustPlatform/rosca-contracts into main_develop_recently_pulled

Commit:
added cleanUpPreviousRound Test and StartRound Test

Commit:
minor fixes to startRoundUnitTest

Commit:
minor fixes to cleanUpPreviousRoundUnitTest.js

Commit:
minor fixes to cleanUpPreviousRoundUnitTest.js

Commit:
fixes according to the spreadsheet

Commit:
each checkPoints are now independent (using a clean contract to test every checkPoint)

Commit:
minor fixes to test files

Commit:
minor typo fixes to test files

Commit:
deleted build files

Commit:
minor fixes and added explanations to some test cases for better readibility

Commit:
minor fixes and added explanations to some test cases for better readibility

Commit:
tweaked the test for picking random unpaid member in goodStanding

Commit:
Merge branch 'develop' of github.com:WeTrustPlatform/rosca-contracts into main_develop_recently_pulled

# Conflicts:
#	test/cleanUpPreviousRoundUnitTest.js

Commit:
Merge branch 'develop' of github.com:WeTrustPlatform/rosca-contracts into main_develop_recently_pulled

# Conflicts:
#	test/cleanUpPreviousRoundUnitTest.js

Commit:
added missing semi-colon

Commit:
add bidUnitTest.js and constructorUnitTest.js

Commit:
removed bidUnitTest.js and constructorUnitTest.js

Commit:
added bidUnitTest.js and constructorUnitTest.js , ready for PR

Commit:
changed ROSCA to ROSCATest

shine2lay committed:
added contributeUnitTest.js and withdrawUnitTest.js

shine2lay committed:
fixes typo and minor bug

shine2lay committed:
fixes typo and minor bug

shine2lay committed:
Merge branch 'develop' of https://github.com/WeTrustPlatform/rosca-contracts into main_develop

# Conflicts:
#	test/startRoundUnitTest.js

shine2lay committed:
WIP

shine2lay committed:
WIP

shine2lay committed:
WIP

Commit:
WIP temp

shine2lay committed:
WIP

Commit:
WIP temp

shine2lay committed:
PR ready

shine2lay committed:
removed mine (since ron's version have mineOneBlock) already

shine2lay committed:
clean Up lines too long

shine2lay committed:
clean Up lines too long

shine2lay committed:
fix hound comment

shine2lay committed:
fixed hound comment #2

Commit:
WIP temp

Commit:
WIP temp

shine2lay committed:
WIP

Commit:
WIP temp

shine2lay committed:
fixes

Commit:
fixes

Commit:
Merge branch 'to_PR' of github.com:shine2lay/rosca-contracts into unifromTestCases

Commit:
Merge branch 'develop' of github.com:WeTrustPlatform/rosca-contracts into unifromTestCases

# Conflicts:
#	test/bidUnitTest.js
#	test/constructorUnitTest.js
#	test/startRoundUnitTest.js
#	test/withdrawUnitTest.js

Commit:
WIP 2 member

shine2lay committed:
WIP

Commit:
WIP 2 member

Commit:
4MemberFullRosca tweaked to include a case where contract balance is less than what the member can withdraw

Commit:
took out some unnecessary comments

Commit:
more fixes

Commit:
more fixes

Commit:
Merge branch 'develop' of github.com:WeTrustPlatform/rosca-contracts into unifromTestCases2

shine2lay committed:
another go at fee problem

shine2lay committed:
all tests passing now

shine2lay committed:
Merge branch 'main_develop' into credit_develop

# Conflicts:
#	test/4MemberFullRoscaTest.js

shine2lay committed:
Merge branch 'develop' of https://github.com/WeTrustPlatform/rosca-contracts into credit_develop

shine2lay committed:
s

shine2lay committed:
ww

shine2lay committed:
fix

Commit:
Merge branch 'to_PR_2' of github.com:shine2lay/rosca-contracts into unifromTestCases2

Commit:
added test case to address issue #61, added different logic for members with debt when calculating totalFees

Commit:
temp WIP

shine2lay committed:
fixed fee Test, ** Things still havent implemented ** 1. not letting user contribute after rosca ends 2. fix the method of taking out fees

shine2lay committed:
fee system have been integrated , all tests except 4MemberFullRoscaTest is working

shine2lay committed:
fixed comments

shine2lay committed:
fixed the way to check delinquency of delinquent winner in contribute()

shine2lay committed:
ignore 4 memberFullROSCA  and feeUnitTest, it still needs some fixing

shine2lay committed:
fixed fullRoscaTest + feeUnitTest

shine2lay committed:
addressed comments

shine2lay committed:
Merge branch 'develop' of https://github.com/WeTrustPlatform/rosca-contracts into credit_develop

# Conflicts:
#	contracts/ROSCA.sol
#	test/4MemberFullRoscaTest.js
#	test/cleanUpPreviousRoundUnitTest.js
#	test/contributeUnitTest.js
#	test/feesUnitTest.js
#	test/startRoundUnitTest.js
#	test/withdrawUnitTest.js

rmerom commented:
Review status: 0 of 6 files reviewed at latest revision, 21 unresolved discussions.

---

*[test/cleanUpPreviousRoundUnitTest.js, line 108 at r2](https://reviewable.io:443/reviews/wetrustplatform/rosca-contracts/89#-Ka4CyRcrcP-D8hvXQ4R:-Ka4CyRcrcP-D8hvXQ4S:b-zb3m8f) ([raw file](https://github.com/wetrustplatform/rosca-contracts/blob/660059344ab3847432d9655c8cd42d8f57cbc381/test/cleanUpPreviousRoundUnitTest.js#L108)):*
> ```JavaScript
>     }));
> 
>     it("checks if random unpaid delinquent member is picked when no bid was placed + only delinquent members are unpaid",
> ```

can you clarify the part after the +?

---

*[test/cleanUpPreviousRoundUnitTest.js, line 113 at r2](https://reviewable.io:443/reviews/wetrustplatform/rosca-contracts/89#-Ka4DBGt8BAF-ckKbhEi:-Ka4DBGt8BAF-ckKbhEj:b2kcye3) ([raw file](https://github.com/wetrustplatform/rosca-contracts/blob/660059344ab3847432d9655c8cd42d8f57cbc381/test/cleanUpPreviousRoundUnitTest.js#L113)):*
> ```JavaScript
>             MEMBER_LIST, SERVICE_FEE_IN_THOUSANDTHS);
> 
>         // in the first 2 rounds, p0 and p2 will win, and in third round
> ```

can we simplify the test by having less participants? (2 or 3 instead of 4)

---

*[test/cleanUpPreviousRoundUnitTest.js, line 123 at r2](https://reviewable.io:443/reviews/wetrustplatform/rosca-contracts/89#-Ka4DIBxR8wmwsQWoRMW:-Ka4CqFBogJ1hXNy-CBR:b-1s7c35) ([raw file](https://github.com/wetrustplatform/rosca-contracts/blob/660059344ab3847432d9655c8cd42d8f57cbc381/test/cleanUpPreviousRoundUnitTest.js#L123)):*
> ```JavaScript
> 
>         utils.increaseTime(ROUND_PERIOD_IN_DAYS * 86400);
>         yield rosca.startRound();
> ```

for sanity check, should we make sure that the winner is a non-delinquent participant here and in the next round?

---

*[test/cleanUpPreviousRoundUnitTest.js, line 127 at r2](https://reviewable.io:443/reviews/wetrustplatform/rosca-contracts/89#-Ka41nk7wD3ulvKuHV_y:-Ka41nk8aXVX0kSQZpnA:bq53p1j) ([raw file](https://github.com/wetrustplatform/rosca-contracts/blob/660059344ab3847432d9655c8cd42d8f57cbc381/test/cleanUpPreviousRoundUnitTest.js#L127)):*
> ```JavaScript
>         utils.increaseTime(ROUND_PERIOD_IN_DAYS * 86400);
> 
>         yield Promise.all([
> ```

is Promise.all needed?

---

*[test/cleanUpPreviousRoundUnitTest.js, line 148 at r2](https://reviewable.io:443/reviews/wetrustplatform/rosca-contracts/89#-Ka42FJOh1ebmqQqgMxZ:-Ka42FJOh1ebmqQqgMx_:b-lsshkf) ([raw file](https://github.com/wetrustplatform/rosca-contracts/blob/660059344ab3847432d9655c8cd42d8f57cbc381/test/cleanUpPreviousRoundUnitTest.js#L148)):*
> ```JavaScript
>         yield Promise.delay(300);
>         assert.isOk(eventFired, "LogRoundFundReleased didn't occur");
>         assert.include(possibleWinner, winnerAddress, "Non eligible member won the pot");
> ```

a general suggestion, not specific for this PR:
in case tests fail, we get the line number of the failure. If it's easy to understand what the error is, we may not need the

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/wetrustplatform/rosca-contracts/89)*
<!-- Sent from Reviewable.io -->

shine2lay committed:
minor fixes

shine2lay commented:
Review status: 0 of 6 files reviewed at latest revision, 21 unresolved discussions.

---

*[test/cleanUpPreviousRoundUnitTest.js, line 108 at r2](https://reviewable.io:443/reviews/wetrustplatform/rosca-contracts/89#-Ka4CyRcrcP-D8hvXQ4R:-Ka4LnCASpGkXVpzNBll:b-896fix) ([raw file](https://github.com/wetrustplatform/rosca-contracts/blob/660059344ab3847432d9655c8cd42d8f57cbc381/test/cleanUpPreviousRoundUnitTest.js#L108)):*
<details><summary><i>Previously, rmerom (Ron Merom) wrote…</i></summary><blockquote>

can you clarify the part after the +?
</blockquote></details>

Done.

---

*[test/cleanUpPreviousRoundUnitTest.js, line 113 at r2](https://reviewable.io:443/reviews/wetrustplatform/rosca-contracts/89#-Ka4DBGt8BAF-ckKbhEi:-Ka4Lo02zBy6QB1YbcN_:b-57hxe9) ([raw file](https://github.com/wetrustplatform/rosca-contracts/blob/660059344ab3847432d9655c8cd42d8f57cbc381/test/cleanUpPreviousRoundUnitTest.js#L113)):*
<details><summary><i>Previously, rmerom (Ron Merom) wrote…</i></summary><blockquote>

can we simplify the test by having less participants? (2 or 3 instead of 4)
</blockquote></details>

changed it to 2 

---

*[test/cleanUpPreviousRoundUnitTest.js, line 123 at r2](https://reviewable.io:443/reviews/wetrustplatform/rosca-contracts/89#-Ka4DIBxR8wmwsQWoRMW:-Ka4Lrb-VlLowDjdSol-:b-1r0vbi) ([raw file](https://github.com/wetrustplatform/rosca-contracts/blob/660059344ab3847432d9655c8cd42d8f57cbc381/test/cleanUpPreviousRoundUnitTest.js#L123)):*
<details><summary><i>Previously, rmerom (Ron Merom) wrote…</i></summary><blockquote>

for sanity check, should we make sure that the winner is a non-delinquent participant here and in the next round?
</blockquote></details>

no longer needed

---

*[test/cleanUpPreviousRoundUnitTest.js, line 127 at r2](https://reviewable.io:443/reviews/wetrustplatform/rosca-contracts/89#-Ka41nk7wD3ulvKuHV_y:-Ka4LtD2YtBooyvJopjl:b-hfpv0l) ([raw file](https://github.com/wetrustplatform/rosca-contracts/blob/660059344ab3847432d9655c8cd42d8f57cbc381/test/cleanUpPreviousRoundUnitTest.js#L127)):*
<details><summary><i>Previously, rmerom (Ron Merom) wrote…</i></summary><blockquote>

is Promise.all needed?
</blockquote></details>

removed

---

*[test/cleanUpPreviousRoundUnitTest.js, line 148 at r2](https://reviewable.io:443/reviews/wetrustplatform/rosca-contracts/89#-Ka42FJOh1ebmqQqgMxZ:-Ka4M-hUk89r0UJ2zFPO:b32wtd) ([raw file](https://github.com/wetrustplatform/rosca-contracts/blob/660059344ab3847432d9655c8cd42d8f57cbc381/test/cleanUpPreviousRoundUnitTest.js#L148)):*
<details><summary><i>Previously, rmerom (Ron Merom) wrote…</i></summary><blockquote>

a general suggestion, not specific for this PR:
in case tests fail, we get the line number of the failure. If it's easy to understand what the error is, we may not need the
</blockquote></details>

ok

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/wetrustplatform/rosca-contracts/89)*
<!-- Sent from Reviewable.io -->
